### PR TITLE
Integrate clients in shoreline repo + Jenkins pipeline

### PR DESCRIPTION
**Pull request by CedricPerotto**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,10 +2,10 @@
 FROM golang:1.15-alpine AS development
 
 ENV GO111MODULE on
-WORKDIR /go/src/github.com/tidepool-org/shoreline
+WORKDIR /go/src/github.com/mdblp/shoreline
```
Comment (ydrisR) [inline Dockerfile:5]: We need to inject the release number to the build process (build.sh).
Look at crew dockerfile  for an example

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                    sh "$WORKSPACE/build.sh"
+                }
+            }
+        }
+        stage('Test ') {
+            steps {
+                echo 'start mongo to serve as a testing db'
+                sh 'docker network create shorelinetest${RUN_ID} && docker run --rm -d --net=shorelinetest${RUN_ID} --name=mongo4shoreline${RUN_ID} mongo:4.2'
+                script {
+                    docker.image('docker.ci.diabeloop.eu/go-build:1.15').inside("--net=shorelinetest${RUN_ID}") {
```
Comment (ydrisR) [inline Jenkinsfile:35]: Oportunity to generate JUnit test report file `go-junit-report`

File: user/marketo/marketo.go [hunk 3]
```diff
@@ -24,6 +25,7 @@ type Manager interface {
 	CreateListMembershipForUser(newUser User)
 	UpdateListMembershipForUser(oldUser User, newUser User)
 	IsAvailable() bool
+	WaitGroup() *sync.WaitGroup
```
Comment (ydrisR) [inline user/marketo/marketo.go:28]: WHat is the purpose of this WaitGroup object?

File: test.sh [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #!/bin/sh -eu
 
-for D in $(find . -name '*_test.go' ! -path './vendor/*' | cut -f2 -d'/' | uniq); do
+for D in $(find . -name '*_test.go' ! -path './vendor/*' -exec dirname {} \; | uniq); do
```
Comment (ydrisR) [inline test.sh]: Add JUnit report

**Review by ydrisR — CHANGES_REQUESTED** [review]
Looks good, just a few remarks
1. oportunity to add JUnit reports for tests
2. need to inject release number to the docker build and build.sh

File: user/marketo/marketo.go (hunk 3 above)
Comment (CedricPerotto) [reply user/marketo/marketo.go:28]: Here only to run tests with race conditions properly

**Review by ydrisR — APPROVED** [review]

**Pull request merged**